ledger/participant-integration-api/src/main/scala/platform/store/dao/JdbcLedgerDao.scala

```diff
@@ -771,17 +771,21 @@ private class JdbcLedgerDao(
 
   private def deduplicationKey(
       commandId: domain.CommandId,
-      submitter: Ref.Party,
-  ): String = commandId.unwrap + "%" + submitter
+      submitters: List[Ref.Party],
+  ): String =
+    if (submitters.length == 1)
+      commandId.unwrap + "%" + submitters.head
+    else
+      commandId.unwrap + "%" + submitters.asInstanceOf[List[String]].sorted.distinct.mkString("%")
```

Comment:
I assume the number of submitters is always supposed to be low, but sorting and deduplicating a linked list still seems a bit wasteful. Would it rather be the case for this to be a `Set` of some sort?

Reply:
For reference, we expect multi-party submissions to only be used for admin operations (e.g., ledger migration). In almost all cases, the list will contain exactly one element - hence the special case (even though `sorted` and `distinct` are special-cased themselves for a one-element list).

For now, the submitters are a`List` in our code base (well, in the new PRs at least). The initial thought was that this list would be pre-sorted by the ledger API server, but the ledger should not have to depend on that. We should probably replace them by a `Set` everywhere (see https://github.com/digital-asset/daml/pull/8026).

Reply:
That idea sounds very sensible.